test/cmdlineTests/model_checker_bmc_loop_iterations/args

```diff
@@ -0,0 +1 @@
+--model-checker-engine bmc --model-checker-bmc-loop-iterations 3
```

Comment:
Can you also add two more tests equals this, but
1- without any argument for iterations
2- with a bogus argument for iterations (like a string "aaa")

And  number 2 also for JSON. That should be it then for test coverage.

Reply:
Done.

> > Nice, looks good! Sorry for being pedantic, but I think we need two more things:
> > 1- A few negative tests. You can choose a few selected tests (really not that many, maybe 2 per for/while/do-while) and add some assertions that should fail so we see it failing.
> > 2- Command line tests, in order to test the new CLI and JSON options. You can use a single Solidity test (from the ones you added) as the `input.sol` and just change the parameters. You can take a look at `test/cmdlineTests/model_checker*` and `test/cmdlineTests/standard_model_checker*` to see how we add the CLI and JSON tests, respectively.
> 
> @leonardoalt No problem. This has to be tested thoroughly.
> 
> I added those tests.
> 
> There is one more bug in error message formatting, that's why code style check didn't pass.

@leonardoalt I couldn't find where that extra whitespace comes from. I changed the test to mitigate that. I don't think it's worth to put in more time searching where and why this whitespace is printed. If you don't agree I'll keep searching. Just let me know.

Reply:
It's unfortunate but we need to fix that, otherwise we need to change the checker script to allow it, which I don't think will go well.
I'll take a look myself as well.

Reply:
@leonardoalt It's here: https://github.com/ethereum/solidity/blob/98501cf7c8488be80670ae24cde63b54cc2d1e6d/test/cmdlineTests/model_checker_bmc_loop_iterations/err#LL10C6-L10C7 `Note: `

Reply:
@leonardoalt It's here: https://github.com/ethereum/solidity/blob/98501cf7c8488be80670ae24cde63b54cc2d1e6d/test/cmdlineTests/model_checker_bmc_loop_iterations/err#LL10C6-L10C7 `Note: `